src/container/amlManager.ts

```diff
@@ -0,0 +1,159 @@
+                        .then((workspaces: Workspace[]) => {
+                            // More than one workspace can have the same name as long as they are in the different resource group.
+                            const counts: Map<string, number> = this.getWorkspaceNameCounts(workspaces);
+                            return workspaces.map((workspace: Workspace) => {
+                                const label: string = counts.get(workspace.name) === 1 ? workspace.name : `${workspace.name} (${Utility.getResourceGroupFromId(workspace.id)})`;
+                                return new AmlWorkspaceQuickPickItem(label, workspace, azureSubscription);
+                            });
+                        }),
+                );
+            }
+
+            const workspaceItems: AmlWorkspaceQuickPickItem[] = await Utility.awaitPromiseArray<AmlWorkspaceQuickPickItem>(workspacePromises, "Azure Machine Learning Workspace");
+            return workspaceItems;
+        } catch (error) {
+            error.message = `Error fetching workspace list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private async selectImage(workspace: Workspace, session: AzureSession): Promise<vscode.QuickPickItem> {
+        const imageItem: vscode.QuickPickItem = await vscode.window.showQuickPick(this.loadAmlImageItems(workspace, session),
+            { placeHolder: "Select Image", ignoreFocusOut: true });
+        return imageItem;
+    }
+
+    private async loadAmlImageItems(workspace: Workspace, session: AzureSession): Promise<vscode.QuickPickItem[]> {
+        try {
+            const modelMgmtEndpoint: string = await this.getModelMgmtEndpoint(workspace);
+            const credentials: ServiceClientCredentials = session.credentials;
+            const client: ServiceClient = new ServiceClient(credentials);
+            const result: HttpOperationResponse<{ value: IImage[] }> = await client.sendRequestWithHttpOperationResponse<{ value: IImage[] }>({
+                method: "GET",
+                baseUrl: modelMgmtEndpoint,
+                pathTemplate: `/api${workspace.id}/images`,
+                queryParameters: { "api-version": Constants.amlApiVersion },
+                serializationMapper: undefined,
+                deserializationMapper: undefined,
+            });
+
+            if (result.body.value === undefined || result.body.value.length === 0) {
+                throw new Error("No image can be found in the workspace.");
+            }
+
+            return result.body.value.map((image: IImage) => {
+                return { label: image.name, description: image.imageLocation };
+            });
+        } catch (error) {
+            error.message = `Error fetching image list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private getWorkspaceNameCounts(workspaces: Workspace[]): Map<string, number> {
+        const counts: Map<string, number> = new Map<string, number>();
```

Comment:
the name of workspaces could be duplicated under ml account?

Reply:
Yes. I have verified and comment is added in https://github.com/Microsoft/vscode-azure-iot-edge/pull/286/files#diff-7dff5585ca5daca8278354dee63e24b3R58.